examples/adwords_axis/src/main/java/adwords/axis/auth/AdvancedCreateCredentialFromScratch.java

```diff
@@ -176,48 +201,86 @@ public static void runExample(
     reportDefinition.setDateRangeType(ReportDefinitionDateRangeType.YESTERDAY);
     reportDefinition.setReportType(ReportDefinitionReportType.CRITERIA_PERFORMANCE_REPORT);
     reportDefinition.setDownloadFormat(DownloadFormat.CSV);
-    // Enable to allow rows with zero impressions to show.
-    reportDefinition.setIncludeZeroImpressions(true);
+    
     reportDefinition.setSelector(reportSelector);
 
+    ReportingConfiguration reportingConfig =
+        new ReportingConfiguration.Builder()
+            // Enable to allow rows with zero impressions to show.
+            .includeZeroImpressions(false)
+            .build();
+
+    session.setReportingConfiguration(reportingConfig);
+
     ReportDownloadResponse response =
         new ReportDownloader(session).downloadReport(reportDefinition);
-    if (response.getHttpStatus() == HttpURLConnection.HTTP_OK) {
-      FileOutputStream fos = new FileOutputStream(new File(reportFile));
-      Streams.copy(response.getInputStream(), fos);
-      fos.close();
-      System.out.println("Report successfully downloaded: " + reportFile);
-    } else {
-      System.out.println("Report was not downloaded. " + response.getHttpStatus() + ": "
-          + response.getHttpResponseMessage());
-    }
+    FileOutputStream fos = new FileOutputStream(new File(reportFile));
+    Streams.copy(response.getInputStream(), fos);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when Streams.copy throws an exception. Either a) close the object returned by *getInputStream()* in a try-finally block or b) close the resource by declaring the object returned by *getInputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).